art/attacks/inference/attribute_inference/true_label_baseline.py

```diff
@@ -0,0 +1,171 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements attribute inference attacks.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+from typing import Optional, Union, TYPE_CHECKING
+
+import numpy as np
+from sklearn.neural_network import MLPClassifier
+
+from art.estimators.classification.classifier import ClassifierMixin
+from art.attacks.attack import AttributeInferenceAttack
+from art.utils import check_and_transform_label_format, float_to_categorical, floats_to_one_hot
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class AttributeInferenceBaselineTrueLabel(AttributeInferenceAttack):
+    """
+    Implementation of a baseline attribute inference, not using a model.
+
+    The idea is to train a simple neural network to learn the attacked feature from the rest of the features, and the
+    true label. Should be used to compare with other attribute inference results.
+    """
+
+    _estimator_requirements = ()
+
+    def __init__(
+        self,
+        attack_model: Optional["CLASSIFIER_TYPE"] = None,
```

Comment:
Why does this attack only support classifiers? Should be both (same as the regular black-box attack)

Reply:
The `attack_model` parameter is for the model that the baseline attack trains, it must be a classifier because its output is the predicted attribute

Reply:
Oh, right. Ok.